embed-code-go/cli/cli.go

```diff
@@ -282,17 +267,174 @@ func parseListArgument(listArgument string) []string {
 // configFilePath — a path to a yaml configuration file.
 //
 // Returns a filled ConfigFields struct.
-func readConfigFields(configFilePath string) ConfigFields {
+func readConfigFields(configFilePath string) Config {
 	content, err := os.ReadFile(configFilePath)
 	if err != nil {
 		panic(err)
 	}
 
-	configFields := ConfigFields{}
+	configFields := Config{}
 	err = yaml.Unmarshal(content, &configFields)
 	if err != nil {
 		panic(err)
 	}
 
 	return configFields
 }
+
+// Validates if mode is set to check, embed, or analyze.
+func validateMode(mode string) error {
+	isModeSet := isNotEmpty(mode)
+	if !isModeSet {
+		return errors.New("mode must be set")
+	}
+
+	validModes := []string{ModeEmbed, ModeAnalyze, ModeCheck}
+	isValidMode := slices.Contains(validModes, mode)
+
+	if !isValidMode {
+		return fmt.Errorf("invalid value for mode. it must be one of — %s, %s or %s",
+			ModeEmbed, ModeCheck, ModeAnalyze)
+	}
+
+	return nil
+}
+
+// Validates if config is set correctly and does not have mutually exclusive params set.
+func validateIfConfigSetWithFileOrArgs(config Config) error {
```

Comment:
This looks too complicated right now. 
AFAIU we can implement following flow: 
1) If `config file` -> read `run arguments` from file (here we can fail if something is wrong with the file) -> validate `run arguments`;
2) If `run arguments` -> validate `run arguments`. 

We can discuss this. Also, I would suggest thinking about renaming this method to `ValidateConfig`.